machine-learning/tutorials/ShampooSalesAnomalyDetection/Program.cs

```diff
@@ -0,0 +1,120 @@
+﻿// <SnippetAddUsings>
+using System;
+using System.IO;
+using Microsoft.ML;
+// </SnippetAddUsings>
+
+namespace ShampooSalesAnomalyDetection
+{
+    class Program
+    {
+        // <SnippetDeclareGlobalVariables>
+        static readonly string _dataPath = Path.Combine(Environment.CurrentDirectory, "Data", "sales-of-shampoo-over-a-three-ye.csv");
+        //assign the Number of records in dataset file to constant variable
+        const int _docsize = 36;
```

Comment:
Suggestion: Compute the value to allow for changes in the data source.

```csharp
long size = dataView.GetRowCount().HasValue ? dataView.GetRowCount().Value : 0;
```

Reply:
Nice catch! Thanks, @luisquintanilla!